File: Tasks/TasksContext.cs
namespace Rajware.Sample.Tasks;

public class TasksContext {
    private ITaskRepository repo;

    public Task AddTask(string description, DateTime deadline) {
        var newTask = new Task(){
            Description=description,
            Deadline=deadline,
            Completed=false
        };

        var result = repo.Add(newTask);
        return result;
    }

    public List<Task> GetAll() {
        return repo.GetAll();
    }
    public Task GetByID(uint id) {
        var result = repo.GetById(id);
        return result;
    }

    public Task Update(Task task) {
        var result = repo.Update(task);
        return result;
    }

    public Task DeleteByID(uint id) {
        return repo.DeleteByID(id);
    }
    public TasksContext(ITaskRepository repo) {
        this.repo = repo;
    }
}

File: Tasks/TasksContext.vb
Option Strict On
Option Explicit On
Option Infer On

Public Class TasksContext
    Private ReadOnly repo As ITaskRepository

    Public Function AddTask(description As string , deadline As DateTime) As Task
        Dim newTask = New Task() With {
            .Description=description,
            .Deadline=deadline,
            .Completed=False
        }

        Dim result = repo.Add(newTask)
        Return result
    End Function

    Public Function GetAll() As List(Of Task)
        Return repo.GetAll()
    End Function

    Public Function GetByID(id As UInteger) As Task
        Dim result = repo.GetById(id)
        Return result
    End Function

    Public Function Update(task As Task) As Task
        Dim result = repo.Update(task)
        Return result
    End Function

    Public Function DeleteByID(id As UInteger) As Task
        Return repo.DeleteByID(id)
    End Function

    Public Sub New(repo As ITaskRepository)
        Me.repo = repo
    End Sub
End Class